contrib/99designs/gqlgen/tracer_test.go

```diff
@@ -0,0 +1,149 @@
+		handler.Tracer(t),
+	))
+	return c
+}
+
+func TestOptions(t *testing.T) {
+	for name, tt := range map[string]struct {
+		tracerOpts []Option
+		test       func(assert *assert.Assertions, root mocktracer.Span)
+	}{
+		"default": {
+			test: func(assert *assert.Assertions, root mocktracer.Span) {
+				assert.NotNil(root)
+				assert.Equal(root.Tag(ext.ResourceName), "gqlgen.operation")
+				assert.Equal(root.Tag(ext.ServiceName), defaultServiceName)
+				assert.Equal(root.Tag(ext.SpanType), ext.SpanTypeGraphql)
+				assert.Nil(root.Tag(ext.EventSampleRate))
+			},
+		},
+		"WithServiceName": {
+			tracerOpts: []Option{WithServiceName("TodoServer")},
+			test: func(assert *assert.Assertions, root mocktracer.Span) {
+				assert.NotNil(root)
+				assert.Equal("TodoServer", root.Tag(ext.ServiceName))
+			},
+		},
+		"WithAnalytics/true": {
+			tracerOpts: []Option{WithAnalytics(true)},
+			test: func(assert *assert.Assertions, root mocktracer.Span) {
+				assert.NotNil(root)
+				assert.Equal(1.0, root.Tag(ext.EventSampleRate))
+			},
+		},
+		"WithAnalytics/false": {
+			tracerOpts: []Option{WithAnalytics(false)},
+			test: func(assert *assert.Assertions, root mocktracer.Span) {
+				assert.NotNil(root)
+				assert.Nil(root.Tag(ext.EventSampleRate))
+			},
+		},
+		"WithAnalyticsRate": {
+			tracerOpts: []Option{WithAnalyticsRate(0.5)},
+			test: func(assert *assert.Assertions, root mocktracer.Span) {
+				assert.NotNil(root)
+				assert.Equal(0.5, root.Tag(ext.EventSampleRate))
+			},
+		},
+	} {
+		t.Run(name, func(t *testing.T) {
+			assert := assert.New(t)
+			mt := mocktracer.Start()
+			defer mt.Stop()
+			c := newTodoClient(New(tt.tracerOpts...))
+
+			var createResp struct {
+				CreateTodo struct{ ID string }
+			}
+			err := c.Post(`mutation CreateTodo{ createTodo(todo: {text: "todo text"}) {id} }`, &createResp)
+			if err != nil {
+				t.Error(err)
+				return
+			}
+			var root mocktracer.Span
+			for _, span := range mt.FinishedSpans() {
+				if span.ParentID() == 0 {
+					root = span
+				}
+			}
+			assert.NotNil(root)
+			if root != nil {
+				tt.test(assert, root)
+			}
+		})
+	}
+}
+
+func TestResolver(t *testing.T) {
+	assert := assert.New(t)
+	mt := mocktracer.Start()
+	defer mt.Stop()
+	c := newTodoClient(New())
+
+	var createResp struct {
+		CreateTodo struct{ ID string }
+	}
+	operation := "CreateTodo"
+	query := `mutation CreateTodo($text: String!){ createTodo(todo: {text: $text}) {id} }`
+	todoText := "todo text"
+	err := c.Post(query, &createResp, client.Var("text", todoText), client.Operation(operation))
+	if err != nil {
+		t.Error(err)
+		return
+	}
+
+	spans := mt.FinishedSpans()
+	var root mocktracer.Span
+	var resolver mocktracer.Span
+	var field mocktracer.Span
+	for _, span := range spans {
+		switch span.Tag(ext.ResourceName) {
+		case operation:
+			root = span
+		case "MyMutation.createTodo":
+			resolver = span
+		case "Todo.id":
+			field = span
+		}
+	}
+	assert.NotNil(root)
```

Comment:
Can we please check _every_ set property for every span? Operation names, resources and tags?